cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -178,4 +180,42 @@ private String generateMaxTerminationDateParam() {
       .withZone(ZoneId.systemDefault());
     return targetDate.format(formatter);
   }
+
+  private void validateConfiguration() throws MetricRetrievalException {
+    String host = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_HOST);
+    String typeKey = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.TYPE);
+    String maxRetriesKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.MAX_RETRIES);
+    String maxTimeKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.MAX_TIME_SECS);
+    String baseDelayKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.DELAY_BASE_MS);
+    String maxDelayKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.DELAY_MAX_MS);
+
+    if (StringUtils.isEmpty(host)) {
+      throw new MetricRetrievalException("Failed to retrieve metrics for Spark program due to Spark metrics host is " +
```

Comment:
This seems a wrong exception to throw. It is a misconfiguration error, not a retrieval error.